andela/colossus-backend
# #167733811 Add email verification feature

**What does this PR do?**
sets up email verification capability 

**Description of task to be completed**
-add isVerified column to users table and set it to default false
-install nodemailer npm package
-Have the endpoints
POST /api/v1/auth/signup
POST /api/v1/auth/verifyuser  working as expected
-write tests for this feature
-document feature on swagger

**How should this be manually tested?**
once this PR is merged, start your server
-using postman navigate to http://localhost:3000/api/v1/auth/signup and signup with firstName, lastName,password and a VALID email: an invalid email can also be used but you would'nt be able test this feature because you'll need to click  on the link that'l be sent to this email.
-Then check the email, when you click on the link in the mail,  the link opens in a new tab,
copy the **token** in the link, screenshot of the tab that opens is below;  
![Screenshot (158)](https://user-images.githubusercontent.com/46686100/63548735-d1c09880-c526-11e9-8e7d-2f6f2fb40a1b.png)

The link looks like https://localhost:3000//verifyuser?query=${token}

-then in postman send a POST request to
 http://localhost:3000/api/v1/auth/verifyuser?query=pasteTokenHere
**note**
nodemailer needs the folowing env variables to work
GMAIL_USERNAME=a Real gmail Account
GMAIL_PASSWORD= its real password too
also note that if its gmail, you'l need to disable restriction on lessSecureApps on that gmail account
You can do that using this link https://www.google.com/settings/security/lesssecureapps
or you use the email account specifically created for this project;
 andelacolossus19@gmail.com
password: pm me or ferguson
**Any background context**
N/A
**What are the relevant pivotal tracker stories**
#167733811
